src/TraceEvent/TraceEvent.cs

```diff
@@ -2432,6 +2440,11 @@ void ConfirmAllEventsAreInEnumeration()
             if (GetType().Name == "JScriptTraceEventParser")
                 return;
 
+            // TODO FIX NOW ApplicationServerTraceEventParser contains some templates sharing the same event (task) name but different event ids
+            // The validation rule here doesn't quite work for them.
+            if (GetType().Name == "ApplicationServerTraceEventParser")
```

Comment:
The following is an example in  ApplicationServerTraceEventParser that causes the assert failure. The two templates has the same eventName (taskName/opcodeName).

        public event Action<Multidata4TemplateHATraceData> ActivityStateRecord
        {
            add
            {
                // action, eventid, taskid, taskName, taskGuid, opcode, opcodeName, providerGuid, providerName
                source.RegisterEventTemplate(new Multidata4TemplateHATraceData(value, 103, 2609, "WorkflowTracking", WorkflowTrackingTaskGuid, 0, "Info", ProviderGuid, ProviderName));
            }
            remove
            {
                source.UnregisterEventTemplate(value, 103, ProviderGuid);
                source.UnregisterEventTemplate(value, 0, WorkflowTrackingTaskGuid);
            }
        }
        public event Action<Multidata3TemplateHATraceData> ActivityScheduledRecord
        {
            add
            {
                // action, eventid, taskid, taskName, taskGuid, opcode, opcodeName, providerGuid, providerName
                source.RegisterEventTemplate(new Multidata3TemplateHATraceData(value, 104, 2609, "WorkflowTracking", WorkflowTrackingTaskGuid, 0, "Info", ProviderGuid, ProviderName));
            }
            remove
            {
                source.UnregisterEventTemplate(value, 104, ProviderGuid);
                source.UnregisterEventTemplate(value, 0, WorkflowTrackingTaskGuid);
            }
        }